crypto.go

```diff
@@ -26,67 +23,78 @@ func getKey(key string) []byte {
 	}
 	return decodedkey
 }
-func encryptDocs(hits gjson.Result, stream cipher.Stream, fr io.Writer) {
-	var res strings.Builder
-	fmt.Fprintf(&res, "%s\n", hits.Raw)
-	plainText := []byte(res.String())
-	cipherText := make([]byte, len(plainText))
-	stream.XORKeyStream(cipherText, plainText)
-
-	if _, err := io.Copy(fr, bytes.NewReader(cipherText)); err != nil {
-		log.Fatal(err)
-	}
 
+type Encryptor struct {
+	stream cipher.Stream
+	w      io.Writer
+}
+
+type Decryptor struct {
+	stream cipher.Stream
+	r      io.Reader
 }
```

Comment:
These are okay, but just if you haven't seen them https://golang.org/pkg/crypto/cipher/#StreamReader.